haas/api.py

```diff
@@ -510,6 +510,25 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON dictionary of dictionaries, indexed by the name of the network.
+    The interior dictionaries are indexed by project and channel.
+    
+    Example:  {"netA": {"channel": "101", "project": "qproj-01"}}
+    """
+    db = model.Session()
+    networks = db.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access is not None:
+            net = {'channel': n.network_id, 'project': n.access.label}
```

Comment:
Woops - the above statement was from looking at the `networkattachment` table, which is different.

Maybe let's rename `channel` to `driver_id`, since the ID is driver-specific and is *not* a channel. Currently, they are VLANs, but they could be something else in the future. @gsilvis @lokI8 @SahilTikale - what do you think?